can-bus/client.c

```diff
@@ -0,0 +1,117 @@
+/* client.c
+ *
+ * Copyright (C) 2021 wolfSSL Inc.
+ *
+ * This file is part of wolfSSL.
+ *
+ * wolfSSL is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * wolfSSL is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA
+ */
+
+#include "common.h"
+
+#define ISOTP_BUFSIZE 65535
+
+/* The ISO-TP object */
+static IsoTpLink g_link;
+
+/* Alloc send and receive buffer statically in RAM */
+static uint8_t g_isotpRecvBuf[ISOTP_BUFSIZE];
+static uint8_t g_isotpSendBuf[ISOTP_BUFSIZE];
+
+
+int main(int argc, char *argv[])
+{
+    WOLFSSL_CTX *ctx = NULL;
+    WOLFSSL_METHOD* method = NULL;
+    WOLFSSL* ssl = NULL;
+    int ret;
+
+    if (argc != 2) {
+        printf("Usage: ./transmit <CAN interface>\n");
+        return -1;
+    }
+
+    wolfSSL_Init();
+
+    /* Connect to CAN bus provided on command line, filter out everything
+     * except for CAN ID 0x8 which is the server */
+    if (can_connect(argv[1], 0x8)) {
+        return -1;
+    }
+
+    /* Setup ISO-TP to ID 0x7 and set the buffers */
+    isotp_init_link(&g_link, 0x7, g_isotpSendBuf, sizeof(g_isotpSendBuf), g_isotpRecvBuf, sizeof(g_isotpRecvBuf));
+
+    method = wolfTLSv1_2_client_method();
+    if (!method) {
+        printf("Could not init wolfSSL method\n");
+        return -1;
+    }
+
+    ctx = wolfSSL_CTX_new(method);
+    if (!ctx) {
+        printf("Could not init wolfSSL context\n");
+        return -1;
+    }
+
+    /* Set the send and receive callback functions which will use the CAN bus */
+    wolfSSL_CTX_SetIOSend(ctx, send_ssl);
+    wolfSSL_CTX_SetIORecv(ctx, recv_ssl);
+
+    wolfSSL_CTX_set_verify(ctx, WOLFSSL_VERIFY_NONE, NULL);
+
+    if (wolfSSL_CTX_load_verify_locations(ctx, "client.pem", NULL)
+        != SSL_SUCCESS) {
+        fprintf(stderr, "ERROR: failed to load cert, please check the file.\n");
+        return -1;
```

Comment:
Will leak `ctx`. Perhaps a goto with cleanup would be a good option for this example? I realize this may have been a copy paste from another example, but the example should not leak memory in error cases.